models/transformer/transformerbackbone.py

```diff
@@ -62,14 +63,50 @@ def forward(self, x: torch.Tensor, squeeze: bool = True):
         if self.use_sig_backbone:
             x = self.frame_backbone(x)
 
-        if self.mode == 'pretrain_mp':
-            latent, pred = self.autoencoder(x) # TODO: shape?
-            return [pred]
+        if mode == 'pretrain_mp':
+        # Masked Prediction
+            _, pred1, mask1 = self.autoencoder.forward_mask(x, self.mask_ratio)
```

Comment:
why is there 2 times forward and adding losses? I think these two outputs are only needed in contrastive learning. and even there this is done differently (at least in the sleepyco setting)